book/src/ephemeral-local-volumes.md

```diff
@@ -34,25 +34,89 @@ spec:
+the `kubelet` calls _NodeUnpublishVolume_, it is the responsibility of the CSI
+driver to delete the volume.
+
+To support inline, a driver must implement the followings:
 
 - Identity service
 - Node service
 
+### CSI Extension Specification
+
+#### [NodePublishVolume](https://github.com/container-storage-interface/spec/blob/master/spec.md#nodepublishvolume)
+
+##### Arguments
+
+* **`volume_id`**: Volume ID will be created by the Kubernetes and passed to the
+  driver by the kubelet.
+* **`volume_context["csi.storage.k8s.io/ephemeral"]`**: This value will be
+  available and it will be equal to `"true"`.
+
+##### Workflow
+
+The driver will receive the appropriate arguments as defined above when an
+ephemeral volume is requested. The driver will create and publish the volume
+to the specified location as noted in the _NodePublishVolume_ request. Volume
+size and any other parameters required will be passed in verbatim from the
+inline manifest parameters to the `NodePublishVolumeRequest.volume_context`.
+
+There is no guarantee that NodePublishVolume will be called again after a
+failure, regardless of what the failure is. To avoid leaking resources, a CSI
+driver must either always free all resources before returning from
+NodePublishVolume on error or implement some kind of garbage collection.
+
+#### [NodeUnpublishVolume](https://github.com/container-storage-interface/spec/blob/master/spec.md#nodeunpublishvolume)
+
+##### Arguments
+
+No changes
+
+##### Workflow
+
+The driver is responsible of deleting the ephemeral volume once it has
+unpublished the volume. It MAY delete the volume before finishing the request,
+or after the request to unpublish is returned.
+
+## CSIDriver
+
 Kubernetes 1.16 only allows using a CSI driver for an inline volume if
-its [`CSIDriverInfo`](csi-driver-object.md) object explicitly declares
+its [`CSIDriver`](csi-driver-object.md) object explicitly declares
 that the driver supports that kind of usage in its
 `volumeLifecycleModes` field. This is a safeguard against accidentally
 [using a driver the wrong way](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190122-csi-inline-volumes.md#support-for-inline-csi-volumes).
 
-### Feature gates
+## Feature gates
+
 To use inline volume, Kubernetes 1.15 binaries must start with the `CSIInlineVolume` feature gate enabled:
 ```
 --feature-gates=CSIInlineVolume=true
 ```
 
 Kubernetes >= 1.16 no longer needs this as the feature is enabled by default.
 
-### Example implementation
+## References
+
+- [CSI Host Path
+  driver ephemeral volumes support](https://github.com/kubernetes-csi/csi-driver-host-path/blob/9fdddc2061b9013286e01189b2bf3268276af99b/pkg/hostpath/nodeserver.go#L63-L82)
+- [Issue 82507: Drop VolumeLifecycleModes field from CSIDriver API before
+  GA](https://github.com/kubernetes/kubernetes/issues/82507)
+- [Issue 75222: CSI Inline - Update CSIDriver to indicate driver
```

Comment:
I don't see a lot of useful information in this issue that isn't already in the KEP linked above. Maybe the KEP link should be moved down here instead.